src/pages/product/index.ts

```diff
@@ -0,0 +1,142 @@
+            productImages.src = image;
+            productImages.className = 'product-images';
+            productImages.addEventListener('click', () => (productMainImage.src = image));
+            productAdditionalImages.appendChild(productImages);
+        });
+        const productInfo = document.createElement('div');
+        productInfo.className = 'product-info';
+        const pageTitle = document.createElement('h1');
+        pageTitle.innerHTML = currentPet.title;
+        pageTitle.className = 'product-page-title';
+        const productPrice = document.createElement('span');
+        productPrice.innerHTML = `${currentPet.price}$`;
+        productPrice.className = 'price';
+        const productStock = document.createElement('span');
+        productStock.className = 'stock';
+        productStock.innerHTML = this.checkStock(currentPet.stock);
+        const productForm = document.createElement('form');
+        productForm.className = 'product-form';
+        const faq = document.createElement('section');
+        faq.className = 'faq';
+
+        productSection.appendChild(
+            createBreadcrumbs({
+                name: currentPet.title,
+                pet: currentPet.pet,
+                category: currentPet.category,
+                page: routes.CatalogPage,
+            })
+        );
+
+        const addToCart = (e: MouseEvent) => {
+            e.preventDefault();
+            cartStore.increase(currentPet.id);
+            addBtn.style.display = 'none';
+            removeBtn.style.display = 'block';
+        };
+
+        const removeFromCart = (e: MouseEvent) => {
+            e.preventDefault();
+            cartStore.remove(currentPet.id);
+            removeBtn.style.display = 'none';
+            addBtn.style.display = 'block';
+        };
+
+        productSection.appendChild(productCard);
+        productCard.appendChild(imagesGallery);
+        imagesGallery.appendChild(productMainImage);
+        imagesGallery.appendChild(productAdditionalImages);
+        productCard.appendChild(productInfo);
+        productInfo.appendChild(pageTitle);
+        productInfo.appendChild(productPrice);
+        productInfo.appendChild(productStock);
+        productInfo.appendChild(productForm);
+        const addBtn = createButton({ buttonText: 'Add to cart', onClick: addToCart });
+        const removeBtn = createButton({
+            buttonText: 'Drop from cart',
+            classNames: 'remove-btn',
+            onClick: removeFromCart,
+        });
+
+        const isExisted = Boolean(cartStore.getById(currentPet.id));
+
+        if (isExisted) {
+            addBtn.style.display = 'none';
+            removeBtn.style.display = 'block';
+        }
+
+        productForm.appendChild(addBtn);
+        productForm.appendChild(removeBtn);
+        const openModal = (e: MouseEvent) => {
+            e.preventDefault();
+            const isExisted = cartStore.getById(currentPet.id);
```

Comment:
Can you use `isExisted` from line 104?

Reply:
The second `isExisted` is used to get the current data at the time the modal window is opened

Reply:
My bad, sorry. Use separate func for it